Update model_implementation.md (#133)
OceanBioME/OceanBioME.jl: docs/src/model_implementation.md

For this example we are going to implement the simple Nutrient-Phytoplankton model similar to that used in [Chen2015](@citep), although we neglect the nutrient in/outflow terms since they may be added as [boundary conditions](https://clima.github.io/OceananigansDocumentation/stable/model_setup/boundary_conditions/), and modified to conserve nitrogen.

> The first step is to import the abstract type from OceanBioME, some units from Oceananigans (for ease of parameter definition), and [`import`](https://stackoverflow.com/questions/27086159/what-is-the-difference-between-using-and-import-in-julia-when-building-a-mod) some functions from Oceananigans which we will add methods to:

Reviewer: ```suggestion
The first step is to import the abstract type from OceanBioME, some units from Oceananigans (for ease of parameter definition), and [`import`](https://stackoverflow.com/questions/27086159/what-is-the-difference-between-using-and-import-in-julia-when-building-a-mod) some functions from Oceananigans in order to add methods to:
```

end
```

> Here, we use descriptive names for the parameters. Below, each of these will correspond to a symbol (or letter) which is more convenient mathematically and when defining the BGC model functions. In the above code we used `@kwdef` to set default values for the models so that we don't have to set all of these parameters each time we use the model. The default parameter values can optionally be over-ridden by the user when they run the mmodel. We have also included a `sinking_velocity` field in the parameter set to demonstrate how we can get tracers (e.g. detritus) to sink. We also need to define some functions so that OceanBioME and Oceananigans know what tracers and auxiliary fields (e.g. light intensity) we will use:

Reviewer: ```suggestion
Here, we use descriptive names for the parameters. Below, each of these parameters correspond to a symbol (or letter) which is more convenient mathematically and when defining the BGC model functions. In the above code we used `@kwdef` to set default values for the models so that we don't have to set all of these parameters each time we use the model. The default parameter values can optionally be over-ridden by the user when running the model. We have also included a `sinking_velocity` field in the parameter set to demonstrate how we can get tracers (e.g. detritus) to sink. We also need to define some functions so that OceanBioME and Oceananigans know what tracers and auxiliary fields (e.g. light intensity) we will use:
```

biogeochemical_auxiliary_fields(bgc::NutrientPhytoplankton) = biogeochemical_auxiliary_fields(bgc.light_attenuation_model)
```

> Next, we define the functions that specify how the phytoplankton ``P`` evolve. In the absence of advection and diffusion (which will be handled by Oceananigans), we want the phytoplankton to evolve at the rate given by:

Reviewer: ```suggestion
Next, we define the functions that specify how the phytoplankton ``P`` evolve. In the absence of advection and diffusion (both of which are handled by Oceananigans), we want the phytoplankton to evolve at the rate given by:
```

```math
\frac{\partial P}{\partial t} = \mu g(T) f(N) h(PAR) P - mP - bP^2,
```

> where $\mu$ corresponds to the parameter `base_growth_rate`, ``m`` corresponds to the parameter `mortality_rate`, and ``b`` corresponds to the parameter `crowding_mortality_rate`. Here, the functions ``g``, ``f``, and ``h`` are defined by:

Reviewer: ```suggestion
where ``\mu`` is the parameter `base_growth_rate`, ``m`` is the parameter `mortality_rate`, and ``b`` is the parameter `crowding_mortality_rate`. Here, the functions ``g``, ``f``, and ``h`` are:
```

```math
\begin{align}
g(T) &= c_1\exp\left(-c_2|T - T_{opt}|\right),\\
f(N) &= \frac{N}{k_N + N},\\
h(PAR) &= \frac{PAR}{k_P + PAR},
\end{align}
```

> where $c\_1$ corresponds to `temperature_coefficient`,  $c\_2$ corresponds to `temperature_exponent`, ``T_{opt}`` corresponds to `optimal_temperature`, ``k_N`` corresponds to `nutrient_half_saturation`, and ``k_P`` corresponds to `light_half_saturation`. 

Author: Does it still render symbolically without the dollar signs?  I made this change because it appeared as \mu and c_1, etc. in the preview of the .md file

> where $c\_1$ corresponds to `temperature_coefficient`,  $c\_2$ corresponds to `temperature_exponent`, ``T_{opt}`` corresponds to `optimal_temperature`, ``k_N`` corresponds to `nutrient_half_saturation`, and ``k_P`` corresponds to `light_half_saturation`. 

Reviewer: Yeap!
See https://oceanbiome.github.io/OceanBioME.jl/previews/PR133/model_implementation/

> where $c\_1$ corresponds to `temperature_coefficient`,  $c\_2$ corresponds to `temperature_exponent`, ``T_{opt}`` corresponds to `optimal_temperature`, ``k_N`` corresponds to `nutrient_half_saturation`, and ``k_P`` corresponds to `light_half_saturation`. 

Reviewer: (Github’s preview of md files might not work, but Documenter renders this all right!)

> where $c\_1$ corresponds to `temperature_coefficient`,  $c\_2$ corresponds to `temperature_exponent`, ``T_{opt}`` corresponds to `optimal_temperature`, ``k_N`` corresponds to `nutrient_half_saturation`, and ``k_P`` corresponds to `light_half_saturation`. 

Author: Oh, ok, thanks for clarifying.  It was on github where it looked off, but you're right that it is rendering correctly on documenter.

end
```

The first parameter `::Val{:P}` is a special [value type](http://www.jlhub.com/julia/manual/en/function/Val) that allows this function to be dispatched when it is given the value `Val(:P)`. This is how Oceananigans tells the model which forcing function to use. At the start of the `NutrientPhytoplankton` function we unpack some parameters from the model, then calculate each term, and return the total change (the gain minus the loss). 

For this model, the nutrient evolution can be inferred from the rate of change of phytoplankton. Since this is a simple two variable model and the total concentration is conserved, 
```math
\frac{\partial N}{\partial t} = - \frac{\partial P}{\partial t}
```
Hence, we will define the nutrient forcing using as the negative of the phytoplankton forcing
```@example implementing
@inline (bgc::NutrientPhytoplankton)(::Val{:N}, args...) = -bgc(Val(:P), args...)
```

> Finally, we need to define some functions to allow us to update the time-dependent parameters (in this case the PAR and temperature, T):

Reviewer: ```suggestion
Finally, we need to define some functions to allow us to update the time-dependent parameters (in this case the PAR and temperature, ``T``):
```
